plugin/forward/forward.go

```diff
@@ -65,24 +66,36 @@ func (f *Forward) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg
 	}
 
 	fails := 0
-	var span, child ot.Span
-	var upstreamErr error
-	span = ot.SpanFromContext(ctx)
+	i := 0
+	returnErr := errNoHealthy
```

Comment:
I thought a bit about the name `upstreamErr` and I think it's a better name than `returnErr`

Reply:
My intention was to use the same var for upstream error and no healthy error, and get rid of the extra 'if' statement at the end of ServeDNS(). Therefore, I renamed it